### ISSUE TITLE
UseNameSuffix for specific source and target

### ISSUE THREAD

**Issue by Alysaar** [agileobjects/AgileMapper#175]
Hi,
The [documentation](https://agilemapper.readthedocs.io/configuration/Member-Name-Patterns/) says that you can use `UseNameSuffix` and `UseNamePattern` for a specific source and target like so :
```csharp
var anonSource = new { _PriceValue = default(double) };

Mapper.WhenMapping
    .From(anonSource)  // Apply to this anon type's mappings
    .ToANew<Product>() // Apply to Product creations
    .UseNamePattern("^_(.+)Value$"); // Match '_PriceValue' to 'Price'
```

However when I try it it does not seem to be implemented.

Here is an example of code that I expected to work but `UseNameSuffix` is not recognized :
```csharp
Mapper.WhenMapping
                .From<object>()
                .To<object>()
                .UseNameSuffix("test");
```

Am I missing something ?
Thank you.

**Comment by SteveWilkes** [comment]
Hi!

Thanks for raising this - apparently I missed it! You can configure type-specific naming patterns inline, but not using the static API. I'm working on it and it's proving quite tricky, but I'll get it sorted out and will report back.

Thanks again!

Steve

**Comment by SteveWilkes** [comment]
Ok!

That was more complex than maybe it should have been, but it's done! The latest v1.6 branch code includes:

```csharp
var anonSource = new { _PriceValue = default(double) };

Mapper.WhenMapping
    .From(anonSource)  // Apply to this anon type's mappings
    .To<Product>()       // Apply to Product mappings
    .UseNamePattern("^_(.+)Value$"); // Match '_PriceValue' to 'Price'
```

It only exists on `.From<>().To<>()` API - not `.From<>().ToANew<>()`, etc - so you can't have different naming patterns for object creations and updates, for example. 

Thanks again for pointing this out :)

Steve

**Comment by SteveWilkes** [comment]
`UseNamePattern` on the static API is included in v1.6, which is now available [on NuGet](https://www.nuget.org/packages/AgileObjects.AgileMapper/1.6.0).

Thanks again for the feedback!
